File: src/model_v4/entities/SBS_USER.ts
import { Column, Entity, Index, OneToMany, PrimaryColumn } from "typeorm";
import { SBS_PREFS } from "./SBS_PREFS";
import { SBS_TT_ISSUE } from "./SBS_TT_ISSUE";


@Entity("SBS_USER",{schema:"sbsdb"})
@Index("USER_ID",["USER_ID",],{unique:true})
@Index("sbsuser_index1",["ROLLE",])
export class SBS_USER {

    @PrimaryColumn("bigint",{ 
        nullable:false,
        primary:true,
        name:"USER_INDEX"
        })
    USER_INDEX:string;
        

    @Column("varchar",{ 
        nullable:true,
        length:50,
        default: () => "'NULL'",
        name:"NACHNAME"
        })
    NACHNAME:string | null;
        

    @Column("varchar",{ 
        nullable:false,
        length:50,
        name:"PASSWORD"
        })
    PASSWORD:string;
        

    @Column("bigint",{ 
        nullable:false,
        name:"ROLLE"
        })
    ROLLE:string;
        

    @Column("varchar",{ 
        nullable:false,
        unique: true,
        length:50,
        name:"USER_ID"
        })
    USER_ID:string;
        

    @Column("varchar",{ 
        nullable:true,
        length:50,
        default: () => "'NULL'",
        name:"VORNAME"
        })
    VORNAME:string | null;
        

   
    @OneToMany(type=>SBS_TT_ISSUE, SBS_TT_ISSUE=>SBS_TT_ISSUE.uSER_INDEX,{ onDelete: 'RESTRICT' ,onUpdate: 'RESTRICT' })
    sBS_TT_ISSUEs:SBS_TT_ISSUE[];
    

   
    @OneToMany(type=>SBS_PREFS, SBS_PREFS=>SBS_PREFS.uSER_INDEX,{ onDelete: 'RESTRICT' ,onUpdate: 'RESTRICT' })
    sBS_PREFSs:SBS_PREFS[];
    
}

File: SbsdbServer/sbsdbv4/model/SbsUser.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace hb.SbsdbServer.sbsdbv4.model
{
    [Table("SBS_USER")]
    public partial class SbsUser
    {
        public SbsUser()
        {
            SbsPrefs = new HashSet<SbsPrefs>();
            SbsTtIssue = new HashSet<SbsTtIssue>();
        }

        [Column("USER_INDEX", TypeName = "bigint(20)")]
        public long UserIndex { get; set; }
        [Column("NACHNAME", TypeName = "varchar(50)")]
        public string Nachname { get; set; }
        [Required]
        [Column("PASSWORD", TypeName = "varchar(50)")]
        public string Password { get; set; }
        [Column("ROLLE", TypeName = "bigint(20)")]
        public long Rolle { get; set; }
        [Required]
        [Column("USER_ID", TypeName = "varchar(50)")]
        public string UserId { get; set; }
        [Column("VORNAME", TypeName = "varchar(50)")]
        public string Vorname { get; set; }

        [InverseProperty("UserIndexNavigation")]
        public virtual ICollection<SbsPrefs> SbsPrefs { get; set; }
        [InverseProperty("UserIndexNavigation")]
        public virtual ICollection<SbsTtIssue> SbsTtIssue { get; set; }
    }
}
